packages/builder/builder/src/plugins/tsChecker.ts

```diff
@@ -86,7 +88,7 @@ export const builderPluginTsChecker = (): DefaultBuilderPlugin => {
           },
           config.tools.tsChecker,
           undefined,
-          deepMerge,
+          mergeBuilderConfig,
```

Comment:
Can you try this package?

https://www.npmjs.com/package/deepmerge

We plan to replace `lodash.merge` with `deepmerge` in Rsbuild.